kernel/src/process_checking.rs

```diff
@@ -0,0 +1,356 @@
+//! Traits, types, and sample implementations of application credentials
```

Comment:
Should this file be called `process_checker.rs`?

Reply:
I'd like to keep this as the gerund form; it implies that it's the traits for checking. The checkers that are in this file are really only intended to be examples. The expectation is that platforms with particular security requirements will write their own.